### ISSUE TITLE
Fundamental flaw during normalization - illegal use of future information

### ISSUE THREAD

**Issue by Bitcoin-Betting-Official** [AnswerXuan/Galformer-Improved-Transformer-for-time-series-Prediction#3]
def load_data(df, seq_len , mul, normalize=True):
       .....

        # 训练集和测试集归一化
        if normalize:
            standard_scaler = preprocessing.StandardScaler()
            train = standard_scaler.fit_transform(train)
            valid = standard_scaler.fit_transform(valid)
            test = standard_scaler.fit_transform(test)

this code uses future data to normalize the test set.  

It should be instead:  

if normalize:
    standard_scaler = preprocessing.StandardScaler()
    train = standard_scaler.fit_transform(train) 
    valid = standard_scaler.transform(valid)    
    test = standard_scaler.transform(test)

**Comment by AnswerXuan** [comment]
# 训练集和测试集归一化
        if normalize:
            standard_scaler = preprocessing.StandardScaler()
            train = standard_scaler.fit_transform(train)
            valid = standard_scaler.transform(valid)
            test = standard_scaler.transform(test) 

The code has been updated. Thank you for your question.
